debian/clearwater-queue-manager.postinst

```diff
@@ -0,0 +1,104 @@
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version, along with the "Special Exception" for use of
+# the program along with SSL, set forth below. This program is distributed
+# in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details. You should have received a copy of the GNU General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+#
+# The author can be reached by email at clearwater@metaswitch.com or by
+# post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+#
+# Special Exception
+# Metaswitch Networks Ltd  grants you permission to copy, modify,
+# propagate, and distribute a work formed by combining OpenSSL with The
+# Software, or a work derivative of such a combination, even if such
+# copying, modification, propagation, or distribution would otherwise
+# violate the terms of the GPL. You must comply with the GPL in all
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+#
+# see: dh_installdeb(1)
+
+set -e
+
+# summary of how this script can be called:
+#        * <postinst> `configure' <most-recently-configured-version>
+#        * <old-postinst> `abort-upgrade' <new version>
+#        * <conflictor's-postinst> `abort-remove' `in-favour' <package>
+#          <new-version>
+#        * <postinst> `abort-remove'
+#        * <deconfigured's-postinst> `abort-deconfigure' `in-favour'
+#          <failed-install-package> <version> `removing'
+#          <conflicting-package> <version>
+# for details, see http://www.debian.org/doc/debian-policy/ or
+# the debian-policy package
+
+NAME=clearwater-queue-manager
+BASE_DIR=/usr/share/clearwater/clearwater-queue-manager
+
+case "$1" in
+    configure)
+        # Create the specific user.
+        if ! grep -q "^$NAME:" /etc/passwd ; then
+            useradd --system -M -d /nonexistent -s /bin/false $NAME
+        fi
+        # Ensure the log directory exists and is owned by the user.
+        mkdir -p -m 755 /var/log/$NAME
+        chown -R $NAME:root /var/log/$NAME
+
+        virtualenv --python=$(which python) $BASE_DIR/env
+        $BASE_DIR/env/bin/easy_install --allow-hosts=None -f $BASE_DIR/eggs/ $BASE_DIR/eggs/clearwater_queue_manager-1.0-py2.7.egg
+        chown -R $NAME:root $BASE_DIR/env
+
+        mkdir -p $BASE_DIR/plugins
```

Comment:
You can use -m 644 here (as you do on line 62 above) - although I think you want 755, as the execute bit on a directory determines whether you can run `ls`.